File: CanHeGetIn/sol.c
#include<stdio.h>

int main(int argc, char const *argv[])
{
    // input
    int max_seats, first_station, last_station;
    scanf("%d", &max_seats);
    scanf("%d %d", &first_station, &last_station);

    int seats[32];
    int seats_length = 0;
    char c;
    while(scanf("%d%c", &seats[seats_length], &c) == 1) {
        seats_length++;
        if (c == '\n') break;
    }

    // solution

    for (int i = first_station; i <= last_station; i++) {
        if (seats[i] >= 150) {
            printf("false\n");
            return 0;
        }
    }
    printf("true\n");
    return 0;
}

File: CanHeGetIn/sol.pas
procedure sol(max_seats, first_station, last_station: integer; stations : Array of integer);
var i : integer;
begin

    for i:=first_station to last_station do
    begin
        if (stations[i] >= max_seats) then
        begin
            writeln('false');
            Exit();
        end;
    end;

    writeln('true');
end;

var
    max_seats : integer;
    first_station : integer;
    last_station : integer;
    stations : Array of integer;
    input_int : integer;
    i : integer;
begin
    // get input
    read(max_seats);
    read(first_station);
    read(last_station);

    i:=-1;
    repeat
        read(input_int);
        Inc(i);
        setLength(stations, i + 1);
        stations[i] := input_int;
    until EOLN;

    sol(max_seats, first_station, last_station, stations)
end.